src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -4240,13 +4240,13 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
     </Copy>
 
-    <!-- If this project produces reference assemblies *and* copied (possibly transitive)
+    <!-- If this project produces reference assemblies *or* copied (possibly transitive)
          references on this build, subsequent builds of projects that depend on it must
          not be considered up to date, so touch this marker file that is considered an
          input to projects that reference this one. -->
     <Touch Files="$(CopyUpToDateMarker)"
            AlwaysCreate="true"
-           Condition="'$(ProduceReferenceAssembly)' == 'true' and '@(ReferencesCopiedInThisBuild)' != ''" />
+           Condition="'$(ProduceReferenceAssembly)' == 'true' or '@(ReferencesCopiedInThisBuild)' != ''" />
```

Comment:
Isn't that correct, though? In the situation where

ClassLibrary1 -> ClassLibrary2 -> ConsoleApp1

If I build everything and then just change the implementation of ClassLibrary1, if we don't touch the up to date marker, then the project system up to date check will think that ClassLibrary2 is up to date since ClassLibrary1's ref assembly will be older than ClassLibrary2.dll. So it won't build ClassLibrary2, so ClassLibrary1.dll won't get copied and ClassLibrary2's marker won't be touched. So ConsoleApp1 won't be built and will run with the old ClassLibrary1.

Touching ClassLibrary1's marker won't cause ClassLibrary2.dll to be rebuilt (since MSBuild will see it as up to date since it's comparing the ref assembly), it will just cause ClassLibrary1.dll to be copied and the marker to be touched. Which seems correct.